### PR TITLE
Greevils begone

### PR DESCRIPTION
**Pull request by Shushishtok**
- Greevils are actually hidden now when no vector is defined in the requiredState. They are now set to dire team whenever that's the case, are pacifists, and get the invisibility modifier. They also properly change teams back and are stripped from the invisibility when chapters use them again (e.g. CH1, CH6). They cannot be found in the map, not even by a scan.
- Fixed an issue that caused a modifier to accumulate on each unit that talked, each time it talked, which led to potential issues. Instead, now modifiers are removed based on their attached particle IDs.
- Fixed an issue that was randomly found, where CH4 scans minimap events remain on the map when skipping to another section.

### CODE REVIEW

File: game/scripts/vscripts/Tutorial/SetupState.ts [hunk 1]
```diff
@@ -107,7 +107,16 @@ function handleBlockades(state: FilledRequiredState) {
...
-        if (!created) {
+        const shouldGolemBeHidden = unit.GetAbsOrigin() === Vector(0, 0, 0);
+
+        if (shouldGolemBeHidden) {
+            unit.SetTeam(DotaTeam.BADGUYS)
+            unit.AddNewModifier(undefined, undefined, "modifier_invisible", {})
+            setUnitPacifist(unit, false);
+            return
+        }
+        else if (!created) {
```
Comment (RobinKa) [inline game/scripts/vscripts/Tutorial/SetupState.ts:118]: no newline :3

File: game/scripts/vscripts/Dialog.ts [hunk 2]
```diff
@@ -40,8 +41,14 @@ class DialogController {
...
-                ParticleManager.DestroyParticle(particleIndex, false);
-                ParticleManager.ReleaseParticleIndex(particleIndex);
+                if (this.currentLine && this.currentLine.speaker) {
+                    for (const modifier of this.currentLine.speaker.FindAllModifiersByName(modifier_particle_attach.name)) {
+                        if ((modifier as modifier_particle_attach).particleID === particleIndex) {
+                            modifier.Destroy()
+                            break;
+                        }
+                    }
+                }
```
Comment (RobinKa) [inline game/scripts/vscripts/Dialog.ts:51]: Looks shady af but whatever

Comment (Shushishtok) [reply game/scripts/vscripts/Dialog.ts:51]: Explained concept in Discord ;p

File: game/scripts/vscripts/Tutorial/SetupState.ts (hunk 1 above)
Comment (Shushishtok) [reply game/scripts/vscripts/Tutorial/SetupState.ts]: Fixed <3

**Pull request merged**